core/src/main/java/org/apache/iceberg/rest/requests/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,130 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.RESTUtils;
+
+/**
+ * A REST request to create a namespace, with an optional set of properties.
+ * The server might also add properties, such as `last_modified_time` etc.
+ */
+public class CreateNamespaceRequest {
+
+  private Namespace namespace;
+  private Map<String, String> properties;
+
+  public CreateNamespaceRequest() {
+    // Needed for Jackson Deserialization.
+  }
+
+  private CreateNamespaceRequest(Namespace namespace, Map<String, String> properties) {
+    this.namespace = namespace;
+    this.properties = properties;
+    validate();
+  }
+
+  CreateNamespaceRequest validate() {
+    ValidationException.check(namespace != null && !namespace.isEmpty(),
+        "Cannot build a request to create a namespace without specifying the namespace");
+    return this;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public Namespace namespace() {
+    return namespace;
+  }
+
+  public Map<String, String> properties() {
+    return RESTUtils.emptyOrImmutableMap(properties);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("namespace", namespace)
+        .add("properties", properties)
+        .toString();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    CreateNamespaceRequest request = (CreateNamespaceRequest) o;
+    return namespace.equals(request.namespace) && Objects.equals(properties, request.properties);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(namespace, properties);
+  }
+
+  public static class Builder {
+    private Namespace namespace;
+    private ImmutableMap.Builder<String, String> properties;
+
+    private Builder() {
+    }
+
+    public Builder withNamespace(Namespace ns) {
+      this.namespace = ns;
+      return this;
+    }
+
+    public Builder setProperties(Map<String, String> props) {
+      // ImmutableMap.Builder will throw on a null map, so we allow for overwriting
+      // with null.... possibly we should throw instead so people don't use the builder improperly?
+      if (props == null) {
+        properties = null;
+        return this;
+      }
```

Comment:
ImmutableMap.Builder will throw on a null map, so this allows for overwriting with `null`.

Possibly we should throw instead so people don't use the builder improperly?